src/sentry/utils/http.py

```diff
@@ -156,3 +158,26 @@ def is_valid_origin(origin, project=None, allowed=None):
         if parsed.path.startswith(path):
             return True
     return False
+
+
+def is_valid_ip(ip_address, project):
+    """
+    Verify that an IP address is not being blacklisted
+    for the given project.
+    """
+    blacklist = project.get_option('sentry:blacklisted-ips')
+    if not blacklist:
+        return True
+
+    ip_network = IPNetwork(ip_address)
```

Comment:
Yeah, it's obviously not exploding, but it's relatively expensive to convert an ip address into this IPNetwork object. So it can be avoided since the common case will be exact matches, I'd imagine.